test/xbgpu/test_engine.py

```diff
@@ -452,15 +527,34 @@ async def test_xengine_end_to_end(
         packet's timestamp (to be non-zero).
         """
         n_baselines = n_ants * (n_ants + 1) * 2
-        first_accumulation_index = 123
-        n_full_accumulations = 3
-        n_total_accumulations = n_full_accumulations + 1
+
+        # We don't want to start at 0, that's boring. So let's offset this a bit.
+        first_batch_index = 123
+
+        # The arbitrarily-selected number above might not be at a convenient
+        # place. We want to be just before a LCM of the heap accumulation
+        # indices of all the corrprods, so that we can get an incomplete
+        # accumulation as the output first.
+        lowest_common_multiple = lcm(
+            *[corrprod_output.heap_accumulation_threshold for corrprod_output in corrprod_outputs]
+        )
+        advance = lowest_common_multiple - (first_batch_index % lowest_common_multiple)
+        first_batch_index += advance
+
+        max_heap_acc_threshold = max(
+            corrprod_output.heap_accumulation_threshold for corrprod_output in corrprod_outputs
+        )
+        min_full_accumulations = (
+            lcm(*[corrprod_output.heap_accumulation_threshold for corrprod_output in corrprod_outputs])
```

Comment:
This is the third time you've used `[corrprod_output.heap_accumulation_threshold for corrprod_output in corrprod_outputs]`. Things will be more readable if you assign it to a variable once.